src/main/drivers/pwm_output.c

```diff
@@ -380,3 +385,41 @@ void servoDevInit(const servoDevConfig_t *servoConfig)
 }
 
 #endif
+
+
+#ifdef BEEPER
+void pwmWriteBeeper(bool onoffBeep)
+{
+        if(onoffBeep == true){
+            *beeperPwm.ccr = (1000000/freqBeep)/2;
+            beeperPwm.enabled = true;
+        } else {
+            *beeperPwm.ccr = 0;
+            beeperPwm.enabled = false;
+        }
+}
+
+void pwmToggleBeeper(void)
+{
+        pwmWriteBeeper(!beeperPwm.enabled);
+}
+
+void beeperPwmInit(uint16_t frequency)
```

Comment:
I would fold this initialisation into the code of `beeperInit` below. A lot of it (like the init of the pin) is common anyway, so it can be simplified.
(Also, you have to check the result of `IOGetByTag` and `timerGetByTag` to catch cases where they don't find a result.)